src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PointerElementAccess.cs

```diff
@@ -12,12 +8,24 @@ internal sealed partial class LocalRewriter
     {
         public override BoundNode VisitPointerElementAccess(BoundPointerElementAccess node)
         {
-            BoundExpression rewrittenExpression = VisitExpression(node.Expression);
+            BoundExpression rewrittenExpression = LowerReceiverOfPointerElementAccess(node.Expression);
             BoundExpression rewrittenIndex = VisitExpression(node.Index);
 
             return RewritePointerElementAccess(node, rewrittenExpression, rewrittenIndex);
         }
 
+        private BoundExpression LowerReceiverOfPointerElementAccess(BoundExpression receiver)
+        {
+            if (receiver is BoundFieldAccess fieldAccess && fieldAccess.FieldSymbol.IsFixed)
+            {
+                var loweredFieldReceiver = VisitExpression(fieldAccess.ReceiverOpt);
+                fieldAccess = fieldAccess.Update(loweredFieldReceiver, fieldAccess.FieldSymbol, fieldAccess.ConstantValueOpt, fieldAccess.ResultKind, fieldAccess.Type);
```

Comment:
Did this affect some test? If not, can we cover?

Reply:
Ping

---
In reply to: [174869481](https://github.com/dotnet/roslyn/pull/24966#discussion_r174869481) [](ancestors = 174869481)

Reply:
I've added a test for this. (`SimpleFixedBufferNestedFieldClassInit`) It does fail if the change is reverted.

---
In reply to: [175237066](https://github.com/dotnet/roslyn/pull/24966#discussion_r175237066) [](ancestors = 175237066,174869481)

Reply:
Ah, thanks

---
In reply to: [175237341](https://github.com/dotnet/roslyn/pull/24966#discussion_r175237341) [](ancestors = 175237341,175237066,174869481)